src/components/MainNav/ProfileDropdown/ProfileDropdown.js

```diff
@@ -57,33 +59,49 @@ class ProfileDropdown extends Component {
     }, []);
   }
 
+  setInitialState(callback) {
```

Comment:
Remind me, what is the advantage of `setInitialState` over just setting it in the constructor? (I'm not disagreeing, just seeking insight.)

Reply:
@MikeTaylor normally there is no difference but this case is a bit different. We need to reset the state (specifically `HandlerComponent`) in order to unmount the previously mounted modal (if present). Otherwise react will keep reusing the same modal (same instance) over and over again (which means the constructor of the HandlerComponent will never be called again). The consequence of this is that the SP modal would open for the first time but after you close it and try to open it again nothing would happen.

The alternative approach here could be to use a random `key`. For example:

<HandlerComponent stripes={this.props.stripes} key={`key-${Date.now()}`} />